rspec-parameterized.gemspec

```diff
@@ -16,6 +16,7 @@ I was inspired by [udzura's mock](https://gist.github.com/1881139).}
   gem.add_dependency('unparser')
   gem.add_dependency('proc_to_ast')
   gem.add_dependency('binding_ninja', '>= 0.2.3')
+  gem.add_dependency('binding_of_caller')
```

Comment:
I think we can remove both binding_ninja and binding_of_caller from the dependency.
I think it would be better to trigger a `LoadError` at `require` time and notice a warning.

Reply:
Doing that might make your gemspec cleaner.

However instead, gem users need to write the following in `Gemfile`, which is inconvenient.

```ruby
gem "rspec-parameterized"

# for Ruby 3.2+
gem "binding_of_caller"

# for Ruby < 3.2
gem "binding_ninja"
```

They and we will need to manually modify the `Gemfile` when upgrading.

So I would rather remove the dependency on `binding_ninja` and use only `binding_of_caller` instead.

This keeps the gem users's upgrade costs low and keeps the implementation of this gem simple.

What do you think?

Reply:
Actually, I also think it is inconvenient.
But TableSyntax is an optional feature. and `binding_ninja` and `binding_of_caller` are native extension libraries that are unnecessary dependencies for some people.

I think requiring all users to compile them makes it harder for casual users.

Reply:
So How do you think about carving out `TableSyntax` into a other gem? (e.g. `rspec-parameterized-table_syntax` )

# Example. 1
## rspec-parameterized.gemspec
```ruby
gem.add_dependency 'rspec'

# TODO: Write other minimum required dependencies.
```

## rspec-parameterized-table_syntax.gemspec
```ruby
gem.add_dependency 'rspec-parameterized'
gem.add_dependency 'binding_of_caller'
```

# Example. 2 
This is similar to rails and rspec gems

## rspec-parameterized-core.gemspec
```ruby
gem.add_dependency 'rspec'

# TODO: Write other minimum required dependencies.
```

## rspec-parameterized-table_syntax.gemspec
```ruby
gem.add_dependency 'rspec-parameterized-core'
gem.add_dependency 'binding_of_caller'
```

## rspec-parameterized.gemspec
```ruby
gem.add_dependency 'rspec-parameterized-core'
gem.add_dependency 'rspec-parameterized-table_syntax'
```

Reply:
Nice!
It's a better solution.

Reply:
Separating the gems is a good idea, but separating the gems is not mandatory for Ruby 3.2 support. Also, I think Ruby 3.2 users will be inconvenienced until the gems are separated.

Therefore, we do Ruby 3.2 support with minimal changes in this major version and separate the gems in the next major version? (#82)

I have written a patch at 0760ea6 to minimize dependencies for Ruby 3.2 support